http-client/src/main/java/io/airlift/http/client/jetty/JettyIoPool.java

```diff
@@ -101,4 +118,52 @@ public String toString()
                 .add("name", name)
                 .toString();
     }
+
+    // based on ScheduledExecutorScheduler
+    private static class ConcurrentScheduler
+            extends AbstractLifeCycle
+            implements Scheduler
+    {
+        private final int threadsPerScheduler;
+        private final ScheduledExecutorService[] schedulers;
+        private final String threadBaseName;
+
+        public ConcurrentScheduler(int schedulerCount, int threadsPerScheduler, String threadBaseName)
+        {
+            this.schedulers = new ScheduledThreadPoolExecutor[schedulerCount];
+            this.threadsPerScheduler = threadsPerScheduler;
+            this.threadBaseName = threadBaseName;
+        }
+
+        @Override
+        protected void doStart()
+                throws Exception
+        {
+            for (int i = 0; i < schedulers.length; i++) {
+                schedulers[i] = Executors.newScheduledThreadPool(threadsPerScheduler, Threads.daemonThreadsNamed(threadBaseName + "-%s-lock" + i));
+            }
+        }
+
+        @Override
+        protected void doStop()
+                throws Exception
+        {
+            for (int i = 0; i < schedulers.length; i++) {
+                schedulers[i].shutdownNow();
+                schedulers[i] = null;
+            }
+        }
+
+        @Override
+        public Task schedule(Runnable task, long delay, TimeUnit unit)
+        {
+            ScheduledExecutorService scheduler = schedulers[ThreadLocalRandom.current().nextInt(schedulers.length)];
```

Comment:
Why do it this way instead of round robining? RNG should be good enough to be broadly consistent, but why not just ensure equality?

Reply:
Random is much easier because there is no coordination required.  Also, it is a uniform random so it should be even.